lib/printer.js

```diff
@@ -1155,6 +1155,36 @@ function genericPrintNoParens(path, options, print) {
 
         return concat(parts);
 
+    case "TemplateElement":
+        return n.value.raw;
+
+    case "TemplateLiteral":
+        var expressions = [];
+        var parts = [];
+        var i = 0;
+
+        path.each(function(childPath) {
+            expressions.push(print(childPath));
+        }, "expressions")
+
+        path.each(function(childPath) {
```

Comment:
It seems the index is passed to the `each` callback as second argument, so there is no need for your own loop variable.